File: 1189-Maximum-Number-of-Balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        b = text.count("b")
        a = text.count("a")
        l = text.count("l") // 2
        o = text.count("o") // 2
        n = text.count("n")

        return min(b, a, l, o, n)

File: neetcode/1189-Maximum-Number-of-Balloons.ts
function maxNumberOfBalloons(text: string): number {
  const freq: Record<string, number> = {}
  const ballon: string = "balloon"
  let res: number = 0

  for (let i = 0; i < text.length; i++) freq[text[i]] = (freq[text[i]] || 0) + 1

  while (true) {
    for (let i = 0; i < ballon.length; i++) {
      if (freq[ballon[i]]) freq[ballon[i]]--
      else return res
    }

    res++
  }

  return res
};